storagenode/pieces/cache.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package pieces
+
+import (
+	"context"
+	"sync"
+
+	"go.uber.org/zap"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storage"
+)
+
+// StoreWithCache has access to the pieces store and also keeps track of real time
+// totals of space used for all pieces and for all pieces by satellite
+type StoreWithCache struct {
+	Store
+
+	cacheMutex sync.RWMutex
```

Comment:
I may have missed it, but I don't think we use `RLock()` on `cacheMutex` anywhere. This should be a plain `sync.Mutex` instead. Partially to avoid confusion ("but why is it an RWMutex?") but also because Mutex.Lock is faster than RWMutex.Lock; it doesn't need to make the extra check for existing read-locks.

You could also change some places to use `RLock()` instead of `Lock()`, but I don't think the expected level of lock contention or the expected hold durations justify it.